corehq/apps/domain/views/settings.py

```diff
@@ -542,10 +555,14 @@ def page_context(self):
             'form': self.form,
             'alerts': [
                 {
-                    'start_time': ServerTime(alert.start_time).user_time(pytz.timezone(alert.timezone))
-                        .ui_string() if alert.start_time else None,
-                    'end_time': ServerTime(alert.end_time).user_time(pytz.timezone(alert.timezone))
-                        .ui_string() if alert.end_time else None,
+                    'start_time': (
+                        self._convert_server_time_to_user_time(alert.start_time, alert.timezone).ui_string()
+                        if alert.start_time else None
+                    ),
+                    'end_time': (
+                        self._convert_server_time_to_user_time(alert.end_time, alert.timezone).ui_string()
+                        if alert.end_time else None
+                    ),
```

Comment:
Looks like if you add an "edit_url" for each alert here (and reversing it like you do in the `manage_alerts.js` file, then you should be able to do something like this in the `manage_alerts.html` file (I'm not 100% sure about the exact syntax though):

```html
              <a class="btn btn-default" href=edit_url style="float: left; margin-right: 5px;">
                {% trans "Edit" %}
              </a>
```

With this, the JS changes are redundant. Up to you if you want to go this route, but it will make for less code.

Reply:
Yes, I did think about this. I didn't want to mud up a URL in alert data and so I let JS handle that. This definitely would have kept JS simpler though it didn't feel good to have an unexpected url as a part alert data.

Reply:
I see. FWIW, I don't see a problem with including the reverse URL in the alert data, since it's only used to generate the html template, so for me it would make sense to keep all sorts of info pertaining to each alert together. But I realize this is being nitty :sweat_smile:

Reply:
Cool thanks :)